### PR TITLE
Use `libcudf` wheel from PR rather than nightly for `polars-polars` CI test job

### PR DESCRIPTION
**Pull request by brandon-b-miller**
This PR fixes an issue where one `cudf-polars` CI job uses the `pylibcudf` wheel generated from the branch being tested, but pulls a libcudf nightly which can cause issues when introducing cython/c++ changes simultaneously.

### CODE REVIEW

File: ci/test_cudf_polars_polars_tests.sh [hunk 1]
```diff
@@ -24,9 +24,15 @@ rapids-logger "Download wheels"
 RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
 RAPIDS_PY_WHEEL_NAME="cudf_polars_${RAPIDS_PY_CUDA_SUFFIX}" RAPIDS_PY_WHEEL_PURE="1" rapids-download-wheels-from-s3 ./dist
 
+# Download the libcudf built in the previous step
```
Comment (bdice) [inline ci/test_cudf_polars_polars_tests.sh:27]: Let's download libcudf + pylibcudf in one block [like the other script](https://github.com/rapidsai/cudf/blob/a6ca0f0068995e5080e1c8d04410a2a1b9dc8b37/ci/test_wheel_cudf_polars.sh#L27-L29) and also perform one installation step [like the other script](https://github.com/rapidsai/cudf/blob/a6ca0f0068995e5080e1c8d04410a2a1b9dc8b37/ci/test_wheel_cudf_polars.sh#L42-L48).

File: ci/test_cudf_polars_polars_tests.sh [hunk 2]
```diff
@@ -24,13 +24,18 @@ rapids-logger "Download wheels"
...
+    -v \
+    "$(echo ./dist/cudf_polars_${RAPIDS_PY_CUDA_SUFFIX}*.whl)[test]" \
+    "$(echo ./local-libcudf-dep/libcudf_${RAPIDS_PY_CUDA_SUFFIX}*.whl)" \
+    "$(echo ./local-pylibcudf-dep/pylibcudf_${RAPIDS_PY_CUDA_SUFFIX}*.whl)"
 
-rapids-logger "Install pylibcudf"
-python -m pip install ./local-pylibcudf-dep/pylibcudf*.whl
 
-rapids-logger "Install cudf_polars"
 python -m pip install $(echo ./dist/cudf_polars*.whl)
```
Comment (bdice) [inline ci/test_cudf_polars_polars_tests.sh:39]: It looks like this is being installed above. Can this line be deleted now?

**Review by bdice — CHANGES_REQUESTED** [review]

File: ci/test_cudf_polars_polars_tests.sh (hunk 2 above)
Comment (brandon-b-miller) [reply ci/test_cudf_polars_polars_tests.sh]: Nice catch, thanks.

**Review by bdice — APPROVED** [review]

**Pull request merged**